### PR TITLE
Adds a news ticker.

### PR DESCRIPTION
**Pull request by glassnotes**
Adds a simple news ticker to the main page that can showcase things like recent releases, news, new demos, etc.

### CODE REVIEW

File: jinja/index.html [hunk 1]
```diff
@@ -68,6 +69,33 @@ <h5>Learn more <i class="fas fa-angle-double-right"></i></h5>
...
+              </li>
+              <li>
+                <b>[2020-12-15]</b> Learn how to compute quantum volume in our
+                <a href="https://pennylane.ai/qml/demos/quantum_volume.html">
+                latest demo </a>.
+              </li>
+              <li>
+                <b>[2020-12-08]</b> PennyLane and AWS join forces for
+                a <a href="https://www.newswire.ca/news-releases/xanadu-s-pennylane-brings-quantum-machine-learning-to-customers-with-amazon-braket-803945748.html">
+                variational quantum programming framework</a>.
```
Comment (josh146) [inline jinja/index.html:89]: Should we change this wording to be the same as the title of the PR?

**Review by josh146 — CHANGES_REQUESTED** [review]
This is awesome @glassnotes! It will be great to have on the landing page.

While it looks really nice in desktop mode, I noticed two issues in mobile mode:

* Long news items 'wrap around', with additional lines resulting in a separate ticker item. E.g., for the QHACK one, the 2021 wraps over to a new line, and in terms of the animation acts like a separate news item. This is probably minor though.

* Maybe we should hide the 'Latest news' title from mobile view:
  image

**Comment by glassnotes** [comment]
> Long news items 'wrap around', with additional lines resulting in a separate ticker item. E.g., for the QHACK one, the 2021  wraps over to a new line, and in terms of the animation acts like a separate news item. This is probably minor though.

I was able to partially fix this by turning off the word wrap - however this means that part of the items is cut off on small screens. What would be really cool is to make it so that items scroll horizontally if they are too long, but I think that would be complicated as it would affect the rate vertical scrolling (like it would have to wait for horizontal scrolling to finish first). @josh146 do you have any experience with that kind of thing?

>  Maybe we should hide the 'Latest news' title from mobile view:
>  image

This is fixed!

**Comment by josh146** [comment]
> What would be really cool is to make it so that items scroll horizontally if they are too long, but I think that would be complicated as it would affect the rate vertical scrolling (like it would have to wait for horizontal scrolling to finish first). @josh146 do you have any experience with that kind of thing?

Unfortunately I don't! The CSS/JS trickery here is already beyond what I've done before 😆 

In retrospect, maybe the word wrap is the best approach? I think in this particular example it looked a bit weird (because the line just had `2021!`), but in general the wrapped line might be longer so it will look more natural.

> This is fixed!

💪

**Comment by josh146** [comment]
@glassnotes just revisiting this PR (apologies if I let it slip!)

I was wondering if we could 'automate' the ticker to show the three most recent blog posts + summaries?

Perhaps we could even go further, and have a section on the page for the three latest posts?

**Comment by glassnotes** [comment]
@josh146 I forgot about this! 

The issue with it was that I was not able to get it looking nice for variable screen sizes, as well as word wrapping for long titles. (I imagine these are solved problems, I just don't know how to implement it). 

I like the idea of having a section on the page for recent posts; are you thinking kind of like the "featured demos" carousel on the main demo page?

**Comment by josh146** [comment]
> I like the idea of having a section on the page for recent posts; are you thinking kind of like the "featured demos" carousel on the main demo page?

Yep, but somehow automatically updated by grabbing the blog RSS feed (so that we don't constantly have to update it ourselves!)

**Comment by co9olguy** [comment]
> > I like the idea of having a section on the page for recent posts; are you thinking kind of like the "featured demos" carousel on the main demo page?
> 
> Yep, but somehow automatically updated by grabbing the blog RSS feed (so that we don't constantly have to update it ourselves!)

Perhaps as a github action that is run whenever a new PR is merged on the `blog` repo?

**Comment by josh146** [comment]
> Perhaps as a github action that is run whenever a new PR is merged on the blog repo?

I was thinking actually more along the lines of javascript within the landing page, that dynamically fetches and displays the blog content.

One advantage over the github action approach is that it wouldn't require new deploys of the landing page to update it.

**Comment by josh146** [comment]
Update: we could use `jquery.rss` to load the blog RSS and display it on the homepage:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>jquery.rss example</title>

    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/css/bootstrap.min.css"
      rel="stylesheet"
    />
    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.8.10/css/mdb.min.css"
      rel="stylesheet"
    />

    <script src="http://code.jquery.com/jquery-1.6.4.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-rss/3.3.0/jquery.rss.min.js"></script>

    <script>
      jQuery(function ($) {
        $('#rss-feeds').rss('http://pennylane.ai/blog/feeds/rss.xml', {
          entryTemplate:
            '<a href="{url}"><div class="card mb-4"><div class="card-body"><div class="card-title">{date} -- {title}</div>{teaserImage}{shortBodyPlain}...</div></div>',
          layoutTemplate: "<div class='feed-container'>{entries}</div>",
          limit: 3,
          support: false,
        });
      });
    </script>
  </head>

  <body>
    <div class="container">
      <div class="card-group" id="rss-feeds" style="display: flex"></div>
    </div>
  </body>
</html>
```

image

**Comment by co9olguy** [comment]
Yeah, that'd be a nice idea :slightly_smiling_face:

**Comment by glassnotes** [comment]
Closing this now that #20 has been merged :tada:

**Pull request closed without merging**